File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/ContentCategoryController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.content.service.ContentCategoryService;
import com.pinyougou.pojo.TbContentCategory;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/contentCategory")
public class ContentCategoryController {

    @Reference
    private ContentCategoryService contentCategoryService;

    @RequestMapping("/findAll")
    public List<TbContentCategory> findAll() throws Exception {
        return contentCategoryService.findAll();
    }

    @RequestMapping("/findByPage")
    public PageResult<TbContentCategory> findByPage(@RequestBody TbContentCategory contentCategory, int page, int size) throws Exception {
        if (page <= 0) {
            page = 1;
        }
        if (size <= 0) {
            size = 10;
        }
        return contentCategoryService.findByPage(contentCategory, page, size);
    }

    @RequestMapping("/findById")
    public TbContentCategory findById(Long id) throws Exception {
        return contentCategoryService.findById(id);
    }

    @RequestMapping("/save")
    public Result save(@RequestBody TbContentCategory contentCategory) {
        try {
            contentCategoryService.save(contentCategory);
            return new Result(true, "保存成功!");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "保存失败!");
        }
    }

    @RequestMapping("/del")
    public Result del(Long[] ids) {
        try {
            contentCategoryService.delete(ids);
            return new Result(true, "删除成功!");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败!");
        }
    }
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/plugins/angularjs/controller/contentCategoryController.js
//控制器
app.controller('contentCategoryController', function ($scope, $controller, contentCategoryService) {

    //继承baseController
    $controller('baseController', {$scope: $scope});

    //查询参数初始化
    $scope.searchEntity = {};

    //分页请求方法
    $scope.findByPage = function (page, size) {
        contentCategoryService.findByPage(page, size, $scope.searchEntity).success(function (data) {
            $scope.list = data.rows;
            $scope.paginationConf.totalItems = data.total;
        })
    };

    //修改回显数据
    $scope.findById = function (id) {
        contentCategoryService.findById(id).success(function (data) {
            $scope.entity = data;
        })
    };

    //新增/修改保存数据
    $scope.save = function () {
        contentCategoryService.save($scope.entity).success(function (data) {
            if (data.success) {
                $scope.reloadPage();
            } else {
                alert(data.message);
            }
        })
    };

    //删除方法
    $scope.del = function () {
        if (confirm('确定要删除吗？')) {
            contentCategoryService.del($scope.ids).success(function (data) {
                if (data.success) {
                    $scope.reloadPage();
                } else {
                    alert(data.message);
                }
            })
        }
    }
});